app/controllers/student_task_controller.rb

```diff
@@ -13,10 +13,27 @@ def list
 
     # #######Tasks and Notifications##################
     @tasknotstarted = @student_tasks.select(&:not_started?)
+    @latetasks = @student_tasks.select(&:late_tasks?) #Tasks which are Finished but not submitted
     @taskrevisions = @student_tasks.select(&:revision?)
 
     ######## Students Teamed With###################
     @students_teamed_with = StudentTask.teamed_students current_user
+
+    ######## Next Immediate Due Date ###############
+    @next_deadline = self.next_dead_line
+  end
+
+  def next_dead_line
+    ######## Next Immediate Due Date ###############
+    next_deadline = nil
+    @student_tasks = StudentTask.from_user current_user
```

Comment:
`@student_tasks` has already been assigned in `list` method, you can reuse it by passing into `next_deadline`.